### PR TITLE
fix(cli): guard Node 24 EBADF fd-close-on-GC crash and fix get-uri fd leak (#1282)

### LINKED ISSUE #1282: CLI crashes with fatal EBADF fd-close-on-GC after successful generation on Node 24 (regression in 2.100.0)
## Summary

Since `morphir-elm` 2.100.0, running `morphir make` under Node 24 can complete generation successfully and then die with a fatal uncaught exception while the process winds down:

```
Compiled in DEV mode. Follow the advice at https://elm-lang.org/0.19.1/optimize ...
<anonymous_script>:0

[Error: EBADF: Closing file descriptor 22 on garbage collection failed, close] {
  errno: -9,
  code: 'EBADF',
  syscall: 'close'
}

Node.js v24.19.0
```

The process exits 1, so any build system driving the CLI reports failure even though the IR was produced. 2.99.0 does not exhibit this under the same Node version.

## Environment

- morphir-elm: 2.100.0 (regression; 2.99.0 is fine)
- Node.js: v24.19.0 (linux x64, GitHub Actions ubuntu-latest; also reproduced across reruns)
- Invocation: `morphir make` via the finos/morphir-scala Mill build against small Elm projects (the `unit-test-framework` example projects in morphir-scala)

## Evidence that it is per-process, not load-related

We first suspected contention between parallel CLI invocations, but the crash reproduces with fully serialized execution (`-j 1`, one Node process at a time): two independent invocations each crashed the same way in the same run ([failing job with serial execution](https://github.com/finos/morphir-scala/actions/runs/31699644203/job/94446139682)). Across two PRs we saw five consecutive occurrences; earlier occurrences on the same infrastructure were intermittent ([example](https://github.com/finos/morphir-scala/actions/runs/31697070223/job/94437834512)).

## Analysis

`EBADF` on `close` inside a garbage-collection finalizer means a file descriptor was closed twice: something closes an fd directly while a handle object still owns it, and when that handle is collected its finalizer close fails. Recent Node makes that failure a fatal uncaught exception rather than a warning, which is why the crash lands after all useful work is done. The regression window is whatever changed fd handling between 2.99.0 and 2.100.0 in the CLI path exercised by `morphir make`.

## Impact and downstream workaround

finos/morphir-scala CI was fully blocked and has pinned back to 2.99.0 with a renovate exclusion for exactly 2.100.0 (finos/morphir-scala#963), so a fixed later release will be picked up automatically. Happy to test a candidate fix from a branch or prerelease.

### PR DESCRIPTION
**Pull request by DamianReeves**
Closes #1282

## Summary

Since Node 24 (DEP0137 EOL, nodejs/node#58536), a `fs.FileHandle`/`ReadStream` GC'd without explicit close throws a fatal `uncaughtException` with `EBADF`/`close` (`Closing file descriptor X on garbage collection failed, close`) instead of a deprecation warning. The CLI's work has already completed when this fires during wind-down, so the process exits 1 and breaks downstream builds (e.g. morphir-scala Mill). The crash is probabilistic and host-scoped (fd 22 in reported runs, 5 consecutive failures, also with `-j 1`), consistent with GC timing.

## Fix

**1. Guard as safety net (6b4d3f2d):**
- `cli/ebadf-guard.js` (CJS) + `cli2/ebadf-guard.ts` (ESM) — idempotent `uncaughtException` handler that swallows _only_ `code===EBADF && syscall===close` and re-throws otherwise (same shape as downstream NODE_OPTIONS shim in finos/morphir-scala#965).
- Installed at top of every CLI entry point (`cli/*` and `cli2/*`) and `package.json` `files` updated.

**2. Root cause (9cdefc6a):**
- `cli2/get-uri-wrapper.ts`: ensure every `getUri` `ReadStream` is `destroy()`'d in `finally` (both after `fetchUriToJson` and inside `toBuffer`). Without this, a `fs.ReadStream` from `get-uri/file.js` can be GC'd with an open fd.
- `cli2/dependencies.ts`: for `file://` (`FileUrl`/`LocalFile`) bypass `get-uri` entirely via `fs/promises.readFile` + `fileURLToPath`. This avoids `get-uri/file.js`'s `open`→`fstat`→`createReadStream` path that can leak an fd if `fstat` fails or the stream isn't fully consumed.

The guard remains for any remaining edge case (e.g. transitive deps).

## Testing

- `mise run build:cli2` passes; `morphir make` on `tests-integration/reference-model` (50 files) and local-dep projects passes.
- Guard verified: EBADF close swallowed (exit 0), non-EBADF / EBADF+read still crashes (exit 7).
- Worktree: `/tmp/morphir-elm-fix-1282` on `fix/ebadf-guard-1282`.

### CODE REVIEW

**Comment by DamianReeves** [comment]
Addressed Codex P1 (cli2/ebadf-guard.ts:21): guard now checks `message.includes('garbage collection')` so ordinary `EBADF`/`close` double-close errors are not swallowed. Pushed cb844898.

File: cli2/ebadf-guard.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * This guard swallows exactly that error shape and re-throws everything else.
+ * It is idempotent across multiple `import` calls via a global flag.
+ */
+const g = globalThis as unknown as { __morphirEbadfGuardInstalled?: boolean };
+if (!g.__morphirEbadfGuardInstalled) {
+  g.__morphirEbadfGuardInstalled = true;
+  process.on('uncaughtException', (err: unknown) => {
+    const e = err as { code?: string; syscall?: string } | null | undefined;
+    if (e && e.code === 'EBADF' && e.syscall === 'close') {
+      return;
```
[the remark this answers is not in the document]
Comment (DamianReeves) [reply cli2/ebadf-guard.ts]: Addressed in 072b211c: guard now requires `message.includes('garbage collection')` in addition to `EBADF`/`close`, so ordinary `fs.close()` double-close errors (without the GC diagnostic) are no longer swallowed and will correctly fail the build. The GC case `Closing file descriptor X on garbage collection failed` is still swallowed as intended.

**Review by AttilaMihaly — APPROVED** [review]
Looks good! Thanks!
